controller/src/main/java/io/pravega/controller/store/PravegaTablesStoreHelper.java

```diff
@@ -244,6 +249,89 @@ public void invalidateCache(String table, String key) {
                 .whenComplete((r, ex) -> releaseEntries(entries));
     }
 
+    public CompletableFuture<List<Version>> removeValues(final String tableName, Set<String> tableKeys, final String appendValue, long requestId) {
+        Supplier<String> errorMessage = () -> String.format("remove values: on table: %s", tableName);
+
+        // read values and add values
+        return withRetries(() -> removeValuesTables(tableName, appendValue, tableKeys, requestId)
```

Comment:
be careful with this withRetries method here.. with conditional updates the retries should be used judiciously or you can get a write conflict on a retry.